File: DarkOrbit 7.5.3/Net/netty/commands/ClientUIMenuBarItemModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class ClientUIMenuBarItemModule
    {
        public static short ID = 5748;

        public String itemId = "";
        public bool visible = false;
        public ClientUITooltipsCommand toolTip;

        public ClientUIMenuBarItemModule(bool pVisible, ClientUITooltipsCommand pTooltipCommand, String pItemId)
        {
            this.visible = pVisible;
            this.toolTip = pTooltipCommand;
            this.itemId = pItemId;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(11362);
            param1.write(toolTip.write());
            param1.writeUTF(this.itemId);
            param1.writeBoolean(this.visible);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/ClientUIMenuBarItemModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

public class ClientUIMenuBarItemModule
        implements ServerCommand {

    public static int     ID      = 22023;
    public        String  itemId  = "";
    public        boolean visible = false;
    public ClientUITooltipsCommand toolTip;

    public ClientUIMenuBarItemModule(boolean pVisible, ClientUITooltipsCommand pTooltipCommand, String pItemId) {
        this.visible = pVisible;
        this.toolTip = pTooltipCommand;
        this.itemId = pItemId;
    }

    public int getID() {
        return ID;
    }

    public int method_1005() {
        return 3;
    }

    public void write(DataOutputStream out) {
        try {
            out.writeShort(ID);
            this.writeInternal(out);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream out) {
        try {
            out.writeBoolean(this.visible);
            out.writeUTF(this.itemId);
            out.writeShort(-6305);
            this.toolTip.write(out);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}